packages/generator/src/complex-type/namespace.ts

```diff
@@ -33,7 +33,21 @@ function factoryFunction(
 }
 
 function getJsonType(complexType: VdmComplexType): string {
-  return '{ [keys: string]: FieldType }';
+  const unionOfAllPropertyTypes = complexType.properties
+    .map(prop => prop.jsType)
+    .reduce(
+      (uniqueList: string[], item) =>
+        uniqueList.includes(item) ? uniqueList : [...uniqueList, item],
+      ['undefined', 'null']
+    )
+    .sort()
+    .reduce(
+      (typeList, item) =>
+        typeList.length === 0 ? item : typeList + ' | ' + item,
+      ''
+    );
```

Comment:
```suggestion
  const unionOfAllPropertyTypes = [...new Set([...complexType.properties.map(prop => prop.jsType), 'undefined', 'null'])].join(' | ');
```

Reply:
Ah, join is the thing I was looking for. I will change it that it builds FieldType + all the complex types.